lib/api/src/grpc/validate.rs

```diff
@@ -116,6 +116,42 @@ impl Validate for crate::grpc::qdrant::quantization_config_diff::Quantization {
     }
 }
 
+impl Validate for crate::grpc::qdrant::update_collection_cluster_setup_request::Operation {
+    fn validate(&self) -> Result<(), ValidationErrors> {
+        use crate::grpc::qdrant::update_collection_cluster_setup_request::Operation;
+        match self {
+            Operation::MoveShard(op) => op.validate(),
+            Operation::ReplicateShard(op) => op.validate(),
+            Operation::AbortTransfer(op) => op.validate(),
+            Operation::DropReplica(op) => op.validate(),
+        }
+    }
+}
+
+impl Validate for crate::grpc::qdrant::MoveShard {
+    fn validate(&self) -> Result<(), ValidationErrors> {
```

Comment:
mandatory question about code duplication :)

should this validation be promoted into common to be shared between REST & gRPC?

Reply:
I didn't do this because the types are different. But now I realize that I can definitely unify the internals into a function for the common crate.

I'll implement that now. Thanks!